doc/sphinxext/allow_nan.py

```diff
@@ -0,0 +1,98 @@
+from sklearn.utils import all_estimators
+from sklearn.linear_model import LogisticRegression
+from sklearn.preprocessing import OrdinalEncoder, OneHotEncoder
+from sklearn.compose import ColumnTransformer
+from io import StringIO
+from docutils import nodes
+
+from docutils.parsers.rst import Directive
+
+
+class Allow_Nan(Directive):
+    def run(self):
+        names1 = []
+        names2 = []
+        names3 = []
+        names4 = []
+        output = StringIO()
+        output.write(
+            f"List of estimators that allow NaN values for type_filter=cluster : "
+        )
+        for name, Estimator in all_estimators(type_filter="cluster"):
+            if Estimator()._get_tags().get("allow_nan") == True:
```

Comment:
This condition can be simplified to:

```suggestion
            if Estimator()._get_tags().get("allow_nan"):
```

the `== True` in a `if` expression is not needed and omitted when writing idiomatic Python code. This idiom is repeated several times in this PR. They can all be simplified.

Reply:
Note, if one really want to test for the physical equality with `True`, then one should instead write `if x is True` instead, but I don't think this is what is intended here.